app/javascript/packs/components/TaskBoard.js

```diff
@@ -1,28 +1,177 @@
+    editCardId: null,
+  };
+
+  lanesMapping() {
+    return {
+      'newTask': { id: 'new_task', name: 'New', state_event: '' },
+      'inDevelopment': { id: 'in_development', name: 'In Dev', state_event: 'to_development' },
+      'inQa': { id: 'in_qa', name: 'In QA', state_event: 'to_qa' },
+      'inCodeReview': { id: 'in_code_review', name: 'in CR', state_event: 'to_code_review' },
+      'readyForRelease': { id: 'ready_for_release', name: 'Ready for release', state_event: 'to_ready_for_release' },
+      'released': { id: 'released', name: 'Released', state_event: 'release' },
+      'archived': { id: 'archived', name: 'Archived', state_event: 'archive' },
     }
-  ]
-}
+  }
+
+  generateLane(id, title) {
+    const tasks = this.state[id];
+
+    return {
+      id,
+      title,
+      totalCount: tasks ? tasks.meta.totalCount : 'None',
+      cards: tasks ? tasks.items.map((task) => {
+        return {
+          ...task,
+          label: task.state,
+          title: task.name
+        };
+      }) : []
+    }
+  }
+
+  getBoard() {
+    const { lanesMapping } = this;
+    const lanes = Object.keys(lanesMapping()).map(key =>
+      this.generateLane(lanesMapping()[key].id, lanesMapping()[key].name),
+    );
+    return { lanes };
+  }
+
+  loadLines() {
+    const { lanesMapping } = this;
+    const lanes = Object.keys(lanesMapping()).map(key =>
+      this.loadLine(lanesMapping()[key].id),
+    );
+  }
+
+  componentDidMount() {
+    this.loadLines();
+  }
+
+  loadLine(state, page = 1) {
+    this.fetchLine(state, page).then(data => {
+      this.setState({ [state]: data });
+    });
+  }
+
+  fetchLine(state, page = 1) {
+    return Fetch.get(
+      Routes.apiV1TasksPath(
+        decamelize({
+          q: { stateEq: state },
+          page: page,
+          perPage: 10,
+          format: 'json',
+        }),
+      ),
+    ).then(({ data }) => {
+      return data;
+    });
+  }
+
+  onLaneScroll(requestedPage, state) {
+    return this.fetchLine(state, requestedPage).then(({ items }) => {
+      return items.map(task => {
+        return {
+          ...task,
+          label: task.state,
+          title: task.name,
+        };
+      });
+    });
+  }
+
+  handleDragEnd = (cardId, sourceLaneId, targetLaneId) => {
+    const stateEvent = this.lanesMapping()[targetLaneId].stateEvent;
+    Fetch.put(Routes.apiV1TaskPath(cardId, { task: { stateEvent } })).then(
+      () => {
+        this.loadLine(sourceLaneId);
+        this.loadLine(targetLaneId);
+      },
+    );
+  };
+
+  handleAddShow = () => {
+    this.setState({ addPopupShow: true });
+  };
+
+  handleAddClose = (added = false) => {
+    this.setState({ addPopupShow: false });
+    if (added == true) {
+      this.loadLine('newTask');
+    }
+  };
+
+  onCardClick = cardId => {
+    this.setState({ editCardId: cardId });
+    this.handleEditShow();
+  };
+
+  handleEditClose = (edited = '') => {
+    this.setState({ editPopupShow: false, editCardId: null });
```

Comment:
You have two fields which means the same thing. And they always connected. You cant have `editCardId` filled, and `editPopupShow` eq false.
In this case, does it make sense to have TWO fields instead of one?